Yoast/wordpress-seo
# Handle sitemap out of bounds page request

Throws an exception in the sitemap providers when a page is requested that is not in the expected bounds.

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where sitemaps would be shown in the `sitemap_index.xml` but result in a 404 when requested.

## Relevant technical choices:

* Throw an exception when the sitemap page requested is out of bounds
* If an external provider does not implement this all pages can still be requested but will be empty

## Test instructions

This PR can be tested by following these steps:

* Make sure the post/page entries are always present in the sitemap index (even if they don't have any entries) - this is true for all public post types
* Make sure the sitemap can be requested (`/post-sitemap.xml` or `/post-sitemap1.xml`)
* Make sure the page that is not supposed to exist is returning a 404 (`/post-sitemap2.xml`)

This is also implemented for `Taxonomies` and `Authors`

Authors:
* If the authors archive is disabled, the sitemap should return a 404 (`/author-sitemap.xml`)
* It should also be excluded from the index

## Quality assurance


Fixes #11428
